tools/depcheck/src/allowlist.rs

```diff
@@ -92,6 +92,14 @@ pub const EXTRA_EXPERIMENTAL_DEPS: &[&str] = &[
     "icu_personnames",
     "icu_relativetime",
     "icu_compactdecimal",
+    "regex",
+    "lazy_static",
+    "aho-corasick",
+    "regex-syntax",
+    "serde",
+    "icu_locid_transform_data",
+    "icu_properties_data",
+    "serde_derive",
```

Comment:
We probably want to reduce this list. If we can get the regexes compiled with `regex-automata` then I think we can avoid most of these extra deps.